### ISSUE TITLE
COPY fails to mark placements INACTIVE on failed nodes

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#1167]
Previously, `COPY` was marking placements `INACTIVE` if it cannot run on that placement successfully. Currently, it fails to mark placements `INACTIVE` which could easily lead to divergent placements without noticing it. 

I think, we should consider fixing this before 6.1. Steps:

```SQL

SELECT * FROM pg_dist_node;
 nodeid | groupid | nodename  | nodeport | noderack | hasmetadata 
--------+---------+-----------+----------+----------+-------------
      1 |       1 | localhost |     9700 | default  | f
      2 |       2 | localhost |     9701 | default  | f
(2 rows)

-- these are just to replicate tests easily
SET citus.shard_count TO 4;
SET citus.shard_replication_factor TO 2;

CREATE TABLE dist_table (key int);
SELECT create_distributed_table('dist_table', 'key');

-- get the shard id for a value only for displaying purposes
 SELECT get_shard_id_for_distribution_column('dist_table', 1);
 get_shard_id_for_distribution_column 
--------------------------------------
                               102008
(1 row)

-- now stop one of the workers
/usr/local/pgsql/bin/pg_ctl -D /Users/onderkalaci/Documents/data_dir/worker_9700 stop

-- load data using COPY
-- everything looks fine
COPY dist_table FROM STDIN;
Enter data to be copied followed by a newline.
End with a backslash and a period on a line by itself.
>> 1
>> \.
COPY 1

-- select WARNs about the placement that it cannot reach
-- also returns a value
SELECT * FROM dist_table ;
WARNING:  connection error: localhost:9700
DETAIL:  could not connect to server: Connection refused
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 9700?
could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 9700?

WARNING:  connection error: localhost:9700
DETAIL:  could not connect to server: Connection refused
	Is the server running on host "localhost" (::1) and accepting
	TCP/IP connections on port 9700?
could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 9700?

 key 
-----
   1
(1 row)


-- restart the worker node that we've stopped before
 /usr/local/pgsql/bin/pg_ctl -D /Users/onderkalaci/Documents/data_dir/worker_9700 -o "-p 9700" -l   /tmp/logfile_9700  start

-- oops now SELECT returns empty because we have divergent placements
 SELECT * FROM dist_table ;
 key 
-----
(0 rows)

-- placement states are ACTIVE
SELECT * FROM pg_dist_shard_placement WHERE shardid = 102008;
 shardid | shardstate | shardlength | nodename  | nodeport | placementid 
---------+------------+-------------+-----------+----------+-------------
  102008 |          1 |           0 | localhost |     9700 |           1
  102008 |          1 |           0 | localhost |     9701 |           2
(2 rows)


```

**Comment by metdos** [comment]
@sumedhpathak, @ozgune shall we fix this in 6.1?

**Comment by marcocitus** [comment]
Yes, this is critical bug that we would otherwise need to backport.

**Comment by metdos** [comment]
@marcocitus assigning to you because you already created the PR :)

**Comment by sumedhpathak** [comment]
Agreed this should be fixed before 6.1 release
